SCHEMAS.md

```diff
@@ -0,0 +1,203 @@
+requests out to have different types.
+
+On the request you would have an optional (i.e. `field?: string | null;`) property on the request.
+
+Then on the response you could include the the field on the response as a required property (i.e. `field: string | null`).
+
+This approach would allow you to introduce a new field with backwards compatibility immediately whilst you update
+consumers to using the new type whilst also emitting the new property in responses.
+
+## Gotchas
+
+### Request/Response type name clash with defined types
+
+When installing the Schemas using the LOKE cli tool if you have request/response objects that are not reference types 
+(i.e. defined using `properties`, `optionalProperties` , etc) then the LOKE cli will create types for you in the
+following format `MethodNameRequest` and `MethodNameResponse`.
+
+When defining your types (in the `definitions` section) you __MUST NOT__ define a type that matches the above format.
+
+For example if you had the following:
+
+ ```typescript
+ // implementation
+interface CreateCustomerRequest {
+  name: string;
+}
+
+interface Customer {
+  id: string;
+  name: string;
+}
+
+CreateCustomer({orgId: string; request: CreateCustomerRequest}): Promise<Customer>
+
+// schema definition
+{
+  ...,
+  definitions: {
+    Customer: {
+      properties: {
+        id: { type: "string" }
+        name: { type: "string" }
+      }
+    },
+    CreateCustomerRequest: {
+      properties: {
+        name: { type: "string" }
+      }
+    }
+  },
+  methods: {
+    CreateCustomer: {
+      requestTypeDef: {
+        properties: {
+          orgId: {type: "string" }
+          request: { ref: "CreateCustomerRequest" }
+        }
+      },
+      responseTypeDef: { ref: "Customer" }
+    }
+  }
+}
+```
+
+When installing the above with the LOKE cli it will generate the following types:
+
+```typescript
+// types created by definitions
+export Type CreateCustomerRequest = {
+  name: string;
+}
+
+export Type Customer {
+  id: string;
+  name: string;
+}
+
+// request type created by LOKE cli, response type is Customer above
+export Type CreateCustomerRequest = {
+  orgId: string;
+  request: CreateCustomerRequest
+}
+```
+
+As you can see in the types above there are two `CreateCustomerRequest` types and this is invalid. To avoid this
+you __MUST__ ensure that your type names never match the format generated by the LOKE cli.
+
+If a better name for type cannot be determined you __SHOULD__ prefix or suffix the type name to avoid the clash.
+Some examples that have been used are `Rpc` prefix or `Body` suffix but be consistent with existing code.
+
+## Date response type
+
+When using Schemas you __MUST_NOT__ use `Date` types in your request/response objects. Instead use strings formatted to
+iso8601/rfc3339 (i.e. "2023-01-20T14:22:23Z") in your implementation and types and use `{ type: "timestamp" }` in the
```

Comment:
Ah interesting, I think I just made these `"string"` in ordering.
Might have to fix up.